service/history/stateBuilder.go

```diff
@@ -656,14 +656,21 @@ func (b *stateBuilderImpl) scheduleWorkflowTimerTask(event *shared.HistoryEvent,
 	timerTasks := []persistence.Task{}
 	now := time.Unix(0, event.GetTimestamp())
 	timeout := now.Add(time.Duration(msBuilder.GetExecutionInfo().WorkflowTimeout) * time.Second)
+	backoffDuration := backoff.NoBackoff
+	startWorkflowAttribute := event.GetWorkflowExecutionStartedEventAttributes()
+	if startWorkflowAttribute.GetFirstDecisionTaskBackoffSeconds() > 0 {
+		backoffDuration = time.Duration(event.GetWorkflowExecutionStartedEventAttributes().GetFirstDecisionTaskBackoffSeconds()) * time.Second
```

Comment:
nit: can make this a small helper method used by this and history.

Reply:
Removed the history engine logic. So I will keep this as it is